jquery/textSelection.d.ts

```diff
@@ -1,62 +1,173 @@
-                end?: number;
-            }
-        ): JQuery;
+        /**
+         * Get the current cursor position (in UTF-16 code units) in a textarea.
+         *
+         * @param {string} command Command to execute
+         * @param {Object} [options]
+         * @param {Object} [options.startAndEnd=false] Return range of the selection rather than just start
+         * @return {number}
+         */
+        textSelection(command: "getCaretPosition", options?: { startAndEnd?: false }): number;
 
+        /**
+         * Set the current cursor position (in UTF-16 code units) in a textarea.
+         *
+         * @param {string} command Command to execute
+         * @param {Object} options
+         * @param {number} options.start
+         * @param {number} [options.end=options.start]
+         * @return {JQuery}
+         * @chainable
+         */
+        textSelection(command: "setSelection", options: { start: number; end?: number }): this;
+
+        /**
+         * Scroll a textarea to the current cursor position. You can set the cursor
+         * position with #setSelection.
+         *
+         * @param {string} command Command to execute
+         * @param {Object} [options]
+         * @param {string} [options.force=false] Whether to force a scroll even if the caret position
+         *  is already visible.
+         * @return {jQuery}
+         * @chainable
+         */
+        textSelection(command: "scrollToCaretPosition", options: { force?: boolean }): this;
+
+        /**
+         * Register an alternative textSelection API for this element.
+         *
+         * @param {string} command Command to execute
+         * @param {Object} functions Functions to replace. Keys are command names (as in #textSelection,
+         *  except 'register' and 'unregister'). Values are functions to execute when a given command is
+         *  called.
+         */
         textSelection(
-            command: "scrollToCaretPosition",
-            commandOptions: {
-                force?: boolean;
-            }
-        ): JQuery;
+            command: "register",
+            functions: Record<string, (commandOptions: unknown) => any>
+        ): void;
 
+        /**
+         * Unregister the alternative textSelection API for this element (see #register).
+         *
+         * @param {string} command Command to execute
+         */
         textSelection(command: "unregister"): void;
 
-        textSelection(command: "register", commandOptions: Record<string, (...args: any[]) => any>): void;
+        /**
+         * Do things to the selection in the textarea, using a command from the alternative textSelection API for this element.
+         *
+         * @param {string} command Command to execute
+         * @param {Mixed} [commandOptions] Options to pass to the command
+         * @return {Mixed} Depending on the command
+         */
+        textSelection(command: string, commandOptions?: any): void;
```

Comment:
This definition is to allow commands from the alternative textSelection API, but there may be a better solution since this one just allows writing any unknown command.